circuits/cpp/src/aztec3/circuits/kernel/public/common.hpp

```diff
@@ -247,6 +247,65 @@ template <typename KernelInput> void propagate_valid_public_data_reads(KernelInp
     }
 }
 
+/**
+ * @brief Propagates new commitments from this iteration to the circuit output.
+ *
+ * @tparam The type of the kernel input
+ * @param public_kernel_inputs The inputs to this iteration to the kernel circuit.
+ * @param circuit_outputs The circuit outputs to be populated
+ */
+template <typename KernelInput>
+void propagate_new_commitments(KernelInput const& public_kernel_inputs, KernelCircuitPublicInputs<NT>& circuit_outputs)
+{
+    // Get the new commitments
+    const auto& public_call_public_inputs = public_kernel_inputs.public_call.call_stack_item.public_inputs;
+
+    const auto& new_commitments = public_call_public_inputs.new_commitments;
+    const auto& storage_contract_address = public_call_public_inputs.call_context.storage_contract_address;
+
+    std::array<NT::fr, KERNEL_NEW_COMMITMENTS_LENGTH> siloed_new_commitments{};
+    for (size_t i = 0; i < new_commitments.size(); ++i) {
```

Comment:
Is it desired to insert `0` into the `new_commitments`? From the other propagations, we are only pushing non-zero values. so wondering why we are not doing that here? Same for `propagate_new_l2_to_l1_messages`. 

Thinking that it could have us run into issues with the length assertions if you have kernel->kernel->kernel where the early ones don't insert anything.

Reply:
Ah ye, missed that. Can keep as this, but think the other is a bit more explicit when reading through it.